crates/runtime/src/lib.rs

```diff
@@ -369,6 +372,48 @@ impl<'a, T> GuestPtr<'a, [T]> {
         (0..self.len()).map(move |i| base.add(i))
     }
 
+    /// Attempts to read a raw `*mut [T]` pointer from this pointer, performing
+    /// bounds checks and utf-8 checks.
```

Comment:
Hmm, if we claim we perform UTF-8 checks here, shouldn't this method apply specifically _and_ only to `[u8]`? I cannot see any other type we'd perform Unicode validation for. Unless you meant for this method be applicable to `IovecArray`, etc., in which though we'd still not perform any UTF-8 validation AFAIK.

Reply:
That’s just a copy-paste error in the comment. I’ll fix